crates/matrix-sdk/src/client.rs

```diff
@@ -995,6 +1009,10 @@ impl Client {
     ///   associated with the device_id. Only necessary the first time you login
     ///   with this device_id. It can be changed later.
     ///
+    /// * `idp_id` - If None, the returned URL will be a generic one, where one
+    ///   has to choose the provider. If Some, the URL will redirect to the
+    ///   login for the specified provider.
```

Comment:
You should be more generic for this method like:

```suggestion
    /// * `idp_id` - The optional ID of the identity provider to login with.
```